demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,694 @@
+        if( reportIdInResponse == reportId )
+        {
+            LogInfo( ( "A valid response with reportId %u received from the "
+                       "AWS IoT Device Defender Service.", reportId ) );
+            status = true;
+        }
+        else
+        {
+            LogError( ( "Unexpected reportId found in the response from the AWS"
+                        "IoT Device Defender Service. Expected: %u, Found: %u, "
+                        "Complete Response: %.*s.",
+                        reportIdInResponse,
+                        reportId,
+                        ( int ) defenderResponseLength,
+                        defenderResponse ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static void publishCallback( MQTTContext_t * pMqttContext,
+                             MQTTPacketInfo_t * pPacketInfo,
+                             MQTTDeserializedInfo_t * pDeserializedInfo )
+{
+    DefenderStatus_t status;
+    DefenderTopic_t api;
+    bool validationResult;
+    MQTTPublishInfo_t * pPublishInfo = pDeserializedInfo->pPublishInfo;
+
+    /* Silence compiler warnings about unused variables. */
+    ( void ) pMqttContext;
+
+    /* Handle incoming publish. The lower 4 bits of the publish packet
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        status = Defender_MatchTopic( pPublishInfo->pTopicName,
+                                      pPublishInfo->topicNameLength,
+                                      &( api ),
+                                      NULL,
+                                      NULL );
+
+        if( status == DefenderSuccess )
+        {
+            if( api == DefenderJsonReportAccepted )
+            {
+                /* Check if the response is valid and is for the report we published. */
+                validationResult = validateDefenderResponse( pPublishInfo->pPayload,
+                                                             pPublishInfo->payloadLength );
+
+                if( validationResult == true )
+                {
+                    LogInfo( ( "The defender report was accepted by the service. Response: %.*s.",
+                               ( int ) pPublishInfo->payloadLength,
+                               ( const char * ) pPublishInfo->pPayload ) );
+                    reportStatus = ReportStatusAccepted;
+                }
+            }
+            else if( api == DefenderJsonReportRejected )
+            {
+                /* Check if the response is valid and is for the report we published. */
+                validationResult = validateDefenderResponse( pPublishInfo->pPayload,
+                                                             pPublishInfo->payloadLength );
+
+                if( validationResult == true )
```

Comment:
Nit: You could save a couple lines, if you wanted
```suggestion
                if( validateDefenderResponse( pPublishInfo->pPayload, pPublishInfo->payloadLength ) )
```

Reply:
I'd not do that as it becomes a little difficult to debug when you want to get see the return value of the function without stepping into it.